storm-core/test/jvm/org/apache/storm/messaging/netty/NettyTest.java

```diff
@@ -0,0 +1,382 @@
+            try (IConnection server = context.bind(null, 0);
+                IConnection client = context.connect(null, "localhost", server.getPort(), remoteBpStatus)) {
+                server.registerRecv(mkConnectionCallback(response::set));
+                waitUntilReady(client, server);
+                byte[] messageBytes = reqMessage.getBytes(StandardCharsets.UTF_8);
+
+                client.send(taskId, messageBytes);
+                Map<Integer, Double> taskToLoad = new HashMap<>();
+                taskToLoad.put(1, 0.0);
+                taskToLoad.put(2, 1.0);
+                server.sendLoadMetrics(taskToLoad);
+
+                List<Integer> tasks = new ArrayList<>();
+                tasks.add(1);
+                tasks.add(2);
+                Testing.whileTimeout(Testing.TEST_TIMEOUT_MS,
+                    () -> client.getLoad(tasks).isEmpty(),
+                    sleep());
+                Map<Integer, Load> load = client.getLoad(tasks);
+                assertThat(load.get(1).getBoltLoad(), is(0.0));
+                assertThat(load.get(2).getBoltLoad(), is(1.0));
+                waitForNotNull(response);
+                TaskMessage responseMessage = response.get();
+                assertThat(responseMessage.task(), is(taskId));
+                assertThat(responseMessage.message(), is(messageBytes));
+            }
+        } finally {
+            context.term();
+        }
+    }
+
+    @Test
+    public void testLoad() throws Exception {
+        doTestLoad(basicConf());
+    }
+
+    @Test
+    public void testLoadWithSasl() throws Exception {
+        doTestLoad(withSaslConf(basicConf()));
+    }
+
+    private void doTestLargeMessage(Map<String, Object> stormConf) throws Exception {
+        LOG.info("3 Should send and receive a large message");
+        String reqMessage = StringUtils.repeat("c", 2048000);
+        IContext context = TransportFactory.makeContext(stormConf);
+        try {
+            AtomicReference<TaskMessage> response = new AtomicReference<>();
+            try (IConnection server = context.bind(null, 0);
+                IConnection client = context.connect(null, "localhost", server.getPort(), remoteBpStatus)) {
+                server.registerRecv(mkConnectionCallback(response::set));
+                waitUntilReady(client, server);
+                byte[] messageBytes = reqMessage.getBytes(StandardCharsets.UTF_8);
+
+                client.send(taskId, messageBytes);
+
+                waitForNotNull(response);
+                TaskMessage responseMessage = response.get();
+                assertThat(responseMessage.task(), is(taskId));
+                assertThat(responseMessage.message(), is(messageBytes));
+            }
+        } finally {
+            context.term();
+        }
+    }
+
+    private Map<String, Object> largeMessageConf() {
+        Map<String, Object> conf = basicConf();
+        conf.put(Config.STORM_MESSAGING_NETTY_BUFFER_SIZE, 1024000);
```

Comment:
Looks like this was 102400, as `test-large-msg` defines.

Reply:
Nice catch